src/emqx_http_lib.erl

```diff
@@ -76,23 +76,24 @@ uri_encode(URI) when is_binary(URI) ->
 -spec uri_parse(string() | binary()) -> {ok, uri_map()} | {error, any()}.
 uri_parse(URI) ->
     try
-        {ok, do_parse(uri_string:normalize(URI))}
+        %% ensure we return string() instead of binary() in uri_map() values.
+        URI1 = maybe_add_default_scheme(unicode:characters_to_list(URI)),
+        {ok, do_parse(uri_string:normalize(URI1))}
     catch
         throw : Reason ->
             {error, Reason}
     end.
 
 do_parse({error, Reason, Which}) -> throw({Reason, Which});
 do_parse(URI) ->
-    %% ensure we return string() instead of binary() in uri_map() values.
-    Map = uri_string:parse(unicode:characters_to_list(URI)),
-    case maps:is_key(scheme, Map) of
```

Comment:
uri_string parses "localhost:80" to #{path => "80",scheme => "localhost"}, as URI like "mailto:John.Doe@example.com" are valid (RFC 3986).
Thus, relying on `maps:is_key(scheme, Map)` is not enough, if really want to add "http://" scheme by default.